### PR TITLE
Add script to generate override snapshot

### PR DESCRIPTION
**Pull request by creydr**
Script to generate the override snapshot based on the CSV file

### CODE REVIEW

File: hack/generate/override-snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+# shellcheck disable=SC1091,SC1090
+source "$(dirname "${BASH_SOURCE[0]}")/../lib/__sources__.bash"
+
+function add_component {
+  local component image_ref parameters git_repo revision dockerfile
+  component=${2}
+  image_ref=${3}
+
+  parameters="$(cosign download attestation "${image_ref}" | jq -r '.payload' | base64 -d | jq -c '.predicate.invocation.parameters')"
```
Comment (creydr) [inline hack/generate/override-snapshot.sh:13]: not sure if skopeo also supports something like this (I didn't see something so far)

File: .konflux/override-snapshot.yaml [hunk 2]
```diff
@@ -0,0 +1,268 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Snapshot
+metadata:
+  name: override-snapshot
```
Comment (pierDipi) [inline .konflux/override-snapshot.yaml:4]: did you consider replacing name with this `generateName: <application-name>-override-snapshot-` ? 

Even if we don't want that, I think the name should have a prefix/suffix with application name

**Comment by creydr** [comment]
```
Error: no attestations associated with quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/kn-client-kn@sha256:96b9b74ef35c0f8fb32ec3b743e585fb7402fa3da35608ae870f9119ba8a0cef
main.go:62: error during command execution: no attestations associated with quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/kn-client-kn@sha256:96b9b74ef35c0f8fb32ec3b743e585fb7402fa3da35608ae870f9119ba8a0cef
```
Seems attestation images are not pushed, when something in the pipeline fails.

/hold

**Comment by creydr** [comment]
Also sometimes the repourl contains the `.git` suffix and sometimes not. e.g.

```
https://github.com/openshift-knative/client

vs

https://github.com/openshift-knative/client.git
```

File: .konflux/override-snapshot.yaml
[the remark this answers is not in the document]
Comment (mgencur) [reply .konflux/override-snapshot.yaml:1]: I thought we want to create "override snapshot" for a specific snapshot that will be "release candidate". If we create this automatically then we'll need to know from prod/eng which snapshot is the release candidate. 
I also think that regular snapshots are generated automatically. Not sure why we want always up-to-date override snapshot.

File: hack/lib/vars.bash [hunk 3]
```diff
@@ -129,8 +129,6 @@ if [ "${USE_RELEASE_NEXT}" = "true" ]; then
   export USE_ARTIFACTS_RELEASE_BRANCH="release-next"
 fi
 
-echo "Branch and Tag: ${USE_ARTIFACTS_RELEASE_BRANCH} - ${USE_IMAGE_RELEASE_TAG}"
-
```
[the remark this answers is not in the document]
Comment (creydr) [reply hack/lib/vars.bash]: restored in 3d970b0c60cdb6310c9032851a492faa9f22beb6

**Comment by creydr** [comment]
> ```
> Error: no attestations associated with quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/kn-client-kn@sha256:96b9b74ef35c0f8fb32ec3b743e585fb7402fa3da35608ae870f9119ba8a0cef
> main.go:62: error during command execution: no attestations associated with quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/kn-client-kn@sha256:96b9b74ef35c0f8fb32ec3b743e585fb7402fa3da35608ae870f9119ba8a0cef
> ```
> 
> Seems attestation images are not pushed, when something in the pipeline fails.
> 
> /hold

As discussed, this is also an indicator, that something is wrong. So failing early here is OK

/unhold

**Pull request merged**